src/blosc2/ndarray.py

```diff
@@ -142,415 +148,611 @@ def _check_allowed_dtypes(
-        kwargs: dict, optional
-            Keyword arguments that are supported by the :func:`empty` constructor.
 
-        Returns
-        -------
-        var_along_axis: np.ndarray or :ref:`NDArray` or scalar
-            The variance of the elements along the axis.
+def max(ndarr: NDArray | NDField | blosc2.C2Array, axis=None, keepdims=False, **kwargs):
+    """
+    Return the maximum along a given axis.
 
-        References
-        ----------
-        `np.var <https://numpy.org/doc/stable/reference/generated/numpy.var.html>`_
-        """
-        expr = blosc2.LazyExpr(new_op=(self, None, None))
-        return expr.var(axis=axis, dtype=dtype, ddof=ddof, keepdims=keepdims, **kwargs)
+    Parameters
+    ----------
+    ndarr: :ref:`NDArray` or :ref:`NDField` or :ref:`C2Array` or :ref:`LazyExpr`
+        The input array or expression.
+    axis: int or tuple of ints, optional
+        Axis or axes along which to operate. By default, flattened input is used.
+    keepdims: bool, optional
+        If this is set to True, the axes which are reduced are left in the result as
+        dimensions with size one. With this option, the result will broadcast correctly
+        against the input array.
 
-    def prod(self, axis=None, dtype=None, out=None, keepdims=False, **kwargs):
-        """
-        Return the product of array elements over a given axis.
+    Returns
+    -------
+    max_along_axis: np.ndarray or :ref:`NDArray` or scalar
+        The maximum of the elements along the axis.
 
-        Parameters
-        ----------
-        axis: int or tuple of ints, optional
-            Axis or axes along which a product is performed. The default, axis=None,
-            will multiply all the elements of the input array. If axis is negative
-            it counts from the last to the first axis.
-        dtype: np.dtype, optional
-            The type of the returned array and of the accumulator in which the
-            elements are multiplied. The dtype of a is used by default unless a has
-            an integer dtype of less precision than the default platform integer.
-        keepdims: bool, optional
-            If this is set to True, the axes which are reduced are left in the result
-            as dimensions with size one. With this option, the result will broadcast
-            correctly against the input array.
-        kwargs: dict, optional
-            Keyword arguments that are supported by the :func:`empty` constructor.
+    References
+    ----------
+    `np.max <https://numpy.org/doc/stable/reference/generated/numpy.max.html>`_
 
-        Returns
-        -------
-        product_along_axis: np.ndarray or :ref:`NDArray` or scalar
-            The product of the elements along the axis.
+    Examples
+    --------
+    >>> import blosc2
+    >>> import numpy as np
+    >>> data = np.array([[11, 2, 36, 24, 5, 69], [73, 81, 49, 6, 73, 0]])
+    >>> ndarray = blosc2.asarray(data)
+    >>> print("Original data:", data)
```

Comment:
I'd rather print the data in the ndarray as: `print("NDArray data:", ndarray[:])`